scipy/sparse/_construct.py

```diff
@@ -935,6 +935,8 @@ def block_diag(mats, format=None, dtype=None):
         nrows, ncols = a.shape
         if issparse(a):
             a = a.tocoo()
+            if a.ndim == 1:
```

Comment:
I'm not entirely sure why this is needed. We're accessing `.row`, `.col`, and `.data` only, and that behavior works with 1d sparse arrays.